### PR TITLE
Actually remove/destroy bit gameobjects

### PR DESCRIPTION
**Pull request by josephmaxwellhyland**
This should actually destroy the bit game object instead of just making it invisible.

### CODE REVIEW

File: shared/util.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { GameVariable, IUpgrade, IUpgradeEffect, VariableModFunction } from "./types";
 import { allUpgrades } from "./upgrades";
 
-function pluck<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
+export function pluck<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
```
Comment (jvgusch) [inline shared/util.ts:4]: This is for my curiosity: does this method mutate the originally array (in this example `clickyBits`) by removing anything that satisfies the predicate? Then returns the last removed object? Typescript is still a bit new to me, just making sure I'm reading this method correctly.

**Review by jvgusch — DISMISSED** [review]

File: shared/util.ts (hunk 1 above)
Comment (josephmaxwellhyland) [reply shared/util.ts:4]: basically yes! now that I'm looking at it more critically, I'm realizing it actually removes all items that satisfy the predicate, but only returns one of the items (the last one). In the two current use cases, there should only ever be one item in the list that matches the predicate. Would probably be worth clarifying that or maybe fixing it to only remove the first item... anyone have thoughts about that?

**Review by jvgusch — DISMISSED** [review]

**Review by emadsu — APPROVED** [review]
looks good thank you

File: frontend/src/game/scenes/Collect.ts [hunk 2]
```diff
@@ -33,8 +33,8 @@ export class Collect extends Scene {
     newClickyBit.setInteractive();
     newClickyBit.on("pointerdown", () => {
       EventBus.emit("add-bit");
-      newClickyBit.removeFromDisplayList();
-      newClickyBit.removeInteractive();
+      const collectedBit: GameObjects.Text = pluckOne(this.clickyBits, (bitObj) => bitObj === newClickyBit);
+      collectedBit.destroy();
```
[the remark this answers is not in the document]
Comment (josephmaxwellhyland) [reply frontend/src/game/scenes/Collect.ts:37]: I will take a look!

[the remark this answers is not in the document]
Comment (josephmaxwellhyland) [reply frontend/src/game/scenes/Collect.ts:37]: This should be fixed now - updated the `pluckOne` function to properly mutate the array

File: shared/util.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
 import { GameVariable, IUpgrade, IUpgradeEffect, VariableModFunction } from "./types";
 import { allUpgrades } from "./upgrades";
 
-function pluck<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
-  const arrCop: Array<T> = arr;
-  arr = [];
+export function pluckOne<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
+  const arrCop: Array<T> = [...arr];
+  arr.length = 0;
```
Comment (josephmaxwellhyland) [inline shared/util.ts:6]: this should actually modify the array properly

**Review by josephmaxwellhyland — COMMENTED** [review]
Amazing catch @MangoMagoCane - this should be fixed now but if you can test it to make sure that would be excellent

File: shared/util.ts (hunk 3 above)
Comment (jvgusch) [reply shared/util.ts:6]: Any idea why `arr = []` didn't work? I'm not super well-versed in the peculiarities of Javascript but I'm really surprised that didn't reset the array but `arr.length = 0` does.

**Review by jvgusch — APPROVED** [review]

File: shared/util.ts (hunk 3 above)
Comment (josephmaxwellhyland) [reply shared/util.ts:6]: I actually have no idea. I'm guessing `arr = []` actually resets the pointer so it doesn't reference the original array anywhere, whereas `array.length = 0` actually references the original array and mutates it. As with many things in JavaScript, for me it comes down to 🤷‍♂️

Comment (MangoMagoCane) [reply shared/util.ts:6]: Javascript feels like the language of bizarre undefined behavior

File: frontend/src/game/scenes/Collect.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (MangoMagoCane) [reply frontend/src/game/scenes/Collect.ts:37]: If my logging is indicative of the actual issue, then it should be properly fixed.

**Pull request merged**